src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -147,6 +150,9 @@ program::program(engine& engine_ref,
     pm = std::unique_ptr<pass_manager>(new pass_manager(*this));
     prepare_nodes(nodes);
     build_program(is_internal);
+
+    _impls_cache = std::unique_ptr<ImplementationsCache>(new ImplementationsCache(_impls_cache_capacity));
```

Comment:
Also, if impls cache is a part of program, then it won't work for deserialized networks. Import/Export doesn't work properly for dynamic models at the moment, but it will be implemented later, so we need to consider that case too. Maybe networks must store shared_ptr to the same impls cache object instead of moving unique_ptr to program?